presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveCreateExternalTables.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.hive;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import io.prestosql.testing.AbstractTestQueryFramework;
+import io.prestosql.testing.QueryRunner;
+import org.apache.hadoop.fs.Path;
+import org.intellij.lang.annotations.Language;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Optional;
+
+import static com.google.common.io.Files.createTempDir;
+import static com.google.common.io.MoreFiles.deleteRecursively;
+import static com.google.common.io.RecursiveDeleteOption.ALLOW_INSECURE;
+import static io.prestosql.tpch.TpchTable.CUSTOMER;
+import static io.prestosql.tpch.TpchTable.ORDERS;
+import static java.lang.String.format;
+
+public class TestHiveCreateExternalTables
+        extends AbstractTestQueryFramework
+{
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        return HiveQueryRunner.createQueryRunner(
+                ImmutableList.of(ORDERS, CUSTOMER),
+                ImmutableMap.of(),
+                "sql-standard",
+                ImmutableMap.of("hive.non-managed-table-writes-enabled", "true"),
+                Optional.empty());
+    }
+
+    @Test
+    public void testCreateExternalTableWithCTASWritesEnabled()
```

Comment:
Perhaps you could have just one test in `TestHiveIntegrationSmokeTest` that runs the CTAS with both the default/custom query runner and asserts failure or success depending on the configuration. `assertUpdate()` and `assertQuery()` both have signatures that accept a session argument and that session could be obtained from different query runners.

Reply:
This may be confusing, especially since the new query runner would be independent from class-level query runner.
I am leaning towards keeping the test classes separate